lib/linter/linter.js

```diff
@@ -1640,34 +1640,9 @@ class Linter {
 
         const slots = internalSlotsMap.get(this);
         const config = providedConfig || {};
+        const { language, settings = {}, languageOptions: originalLanguageOptions = {} } = config;
         const options = normalizeVerifyOptions(providedOptions, config);
-        const languageOptions = config.languageOptions;
-
-        if (config.language === jslang) {
-            languageOptions.ecmaVersion = normalizeEcmaVersionForLanguageOptions(
-                languageOptions.ecmaVersion
-            );
-
-            // Espree expects this information to be passed in
-            if (isEspree(languageOptions.parser)) {
-                const parserOptions = languageOptions.parserOptions;
-
-                if (languageOptions.sourceType) {
-
-                    parserOptions.sourceType = languageOptions.sourceType;
-
-                    if (
-                        parserOptions.sourceType === "module" &&
-                        parserOptions.ecmaFeatures &&
-                        parserOptions.ecmaFeatures.globalReturn
-                    ) {
-                        parserOptions.ecmaFeatures.globalReturn = false;
-                    }
-                }
-            }
-        }
-
-        const settings = config.settings || {};
+        const languageOptions = language.normalizeLanguageOptions?.(originalLanguageOptions) ?? originalLanguageOptions;
```

Comment:
Thoughts about calling `language.normalizeLanguageOptions()` in the `Config` constructor instead of here? That way, language options would be normalized only once per config, not for each file.